ISICbalance/Services/RequestManager.swift

```diff
@@ -9,28 +9,32 @@
 import Foundation
 import Alamofire
 import SwiftSoup
+import SwiftKeychainWrapper
 
+#warning("ask about problem with [weak self] in Alamofire closures")
```

Comment:
do bloku vždycky `[weak self]` a pokud ho tam potřebuješ ne optional, tak si ho přes `guard let self = self else { return }` můžeš rozbalit. Pokud si to silně chytíš v bloku, tak velice snadno si způsobíš leak v appce

Reply:
To že do bloku [weak self] vím, ale měl jsem v tomto filu problém. Nechtělo mi to volat z bloku další metody přes self?... Vyzkouším použít ten guard. ☺️ Přišlo mi divné, že bych v každém bloku musel unwrappovat.

Reply:
jsout triky jak to vyřešit jinak, ale `self` v bloku musí být vždycky `weak` a je potřeba unwrappovat 😊

Reply:
Já tam mám problém v tom, že self je v hned prvním bloku nil. Takže pokud to unwrappnu, tak mi to nepomůže, protože na tom guardu to hned zavolá return 🤔

Reply:
aha 😊 někde ti to chcípne dřív, než se zavolá ten callback. Musíš si to někde silně chytnou, pravděpodobně si vytvoříš instanci uvnitř metody, ale díky tomu bude instance žít pouze po dobu vykonávání té metody.

Reply:
Měl jsem tam hloupou chybu. Vytvářel jsem si toho managera pokaždé při kliknutí na tlačítko... 🤦‍♂️